diff --git a/0_hello_world.c b/0_hello_world.c
index 677909d..1810a44 100644
--- a/0_hello_world.c
+++ b/0_hello_world.c
@@ -120,7 +120,7 @@ int main(int argc, const char *argv[])
 
       logging("Video Codec: resolution %d x %d", pLocalCodecParameters->width, pLocalCodecParameters->height);
     } else if (pLocalCodecParameters->codec_type == AVMEDIA_TYPE_AUDIO) {
-      logging("Audio Codec: %d channels, sample rate %d", pLocalCodecParameters->channels, pLocalCodecParameters->sample_rate);
+      logging("Audio Codec: %d channels, sample rate %d", pLocalCodecParameters->ch_layout.nb_channels, pLocalCodecParameters->sample_rate);
     }
 
     // print its name, id and bitrate
@@ -236,17 +236,17 @@ static int decode_packet(AVPacket *pPacket, AVCodecContext *pCodecContext, AVFra
     if (response >= 0) {
       logging(
           "Frame %d (type=%c, size=%d bytes, format=%d) pts %d key_frame %d [DTS %d]",
-          pCodecContext->frame_number,
+          (int)pCodecContext->frame_num,
           av_get_picture_type_char(pFrame->pict_type),
           pFrame->pkt_size,
           pFrame->format,
           pFrame->pts,
           pFrame->key_frame,
-          pFrame->coded_picture_number
+          (int)pFrame->pkt_dts
       );
 
       char frame_filename[1024];
-      snprintf(frame_filename, sizeof(frame_filename), "%s-%d.pgm", "frame", pCodecContext->frame_number);
+      snprintf(frame_filename, sizeof(frame_filename), "%s-%d.pgm", "frame", (int)pCodecContext->frame_num);
       // Check if the frame is a planar YUV 4:2:0, 12bpp
       // That is the format of the provided .mp4 file
       // RGB formats will definitely not give a gray image
